.github/workflows/main.yml

```diff
@@ -30,23 +31,53 @@ jobs:
       - name: Build docker image
         run: just build
 
-      - name: Test docker image
-        run: just smoke-test
+      - name: Smoke test docker image
+        run: just test-smoke
 
       - name: Save docker image
         run: |
-          docker save research-template | gzip > /tmp/research-template.tar.gz
+          docker save research-template | pigz --fast > /tmp/research-template.tar.gz
 
       - name: Upload docker image
         uses: actions/upload-artifact@v4
         with:
             name: research-template-image
             path: /tmp/research-template.tar.gz
+            # Disable compression; the file is already compressed
+            compression-level: 0
 
-  publish:
+  test-docker-image:
```

Comment:
This job seems to have two key stages: "Check packages" and "Check R Studio installation". I don't understand why they're not part of the "build-and-test" job. Could you clarify when a test would be part of "build-and-test" and when a test would be part of  "test-docker-image"?

Reply:
`build-and-test` is now `build` only.